### PR TITLE
Refactor query polling logic for athena queries

### PR DESCRIPTION
**Pull request by victorphoenix3**
Query status polling logic currently polls until the query is not in intermmediate state. It should instead poll until query is not in the terminal state.
Similar to https://github.com/apache/airflow/issues/19877

### CODE REVIEW

**Review by ferruzzi — APPROVED** [review]
You'll need to wait for a committer to approve and merge, but LGTM.  :+1:   Thanks for adding this one.

**Comment by eladkal** [comment]
@victorphoenix3 can you please fix the failing tests?
https://github.com/apache/airflow/runs/5156246799?check_suite_focus=true#step:7:4789

**Comment by victorphoenix3** [comment]
@eladkal I have updated the PR. Please review

File: tests/providers/amazon/aws/operators/test_athena.py
Comment (victorphoenix3) [inline tests/providers/amazon/aws/operators/test_athena.py:75]: 'SUCCESS' is not a state for athena queries, "SUCCEEDED" is. The tests were failing because of this discrepancy

**Comment by eladkal** [comment]
@victorphoenix3 can you please rebase? tests are broken due to issue with moto (was already fixed in https://github.com/apache/airflow/pull/22046 )

**Review by potiuk — APPROVED** [review]

**Pull request merged**
